### PR TITLE
Introduce Ruby Tag Support Updates for v1.3.0 (based on v1.2.0)

### PR DESCRIPTION
**Pull request by kwan3854**
## Summary

This PR introduces the changes for **v1.3.0**, which builds upon the **v1.2.0** release. It adds support for the new standard `<ruby>` tag format while maintaining backward compatibility with legacy formats (`<ruby="value">` and `<r="value">`).

Since the `v1.3.0` branch does not yet exist in the base repository, this PR is currently targeting the **main branch**. However, it is recommended to create a new branch (e.g., `release/v1.3.0` or `feature/v1.3.0`) based on the `v1.2.0` commit to better track version history.

---

## Changes

1. **New Features**:
   - Added support for standard `<ruby>` tag syntax:
     - Example: `<ruby>fire<rt>ice</rt></ruby>`
   - Rich text support for ruby annotations:
     - Example: `<ruby>fire<rt><color="blue">ice</color></rt></ruby>`
   - Backward compatibility for legacy ruby tag formats:
     - `<ruby="value">fire</ruby>`
     - `<r="value">fire</r>`

2. **Documentation Updates**:
   - Updated the RubyTextMeshPro documentation to explain new tag formats and their usage.

3. **Codebase Improvements**:
   - Refactored the regex parser to handle both legacy and new formats.
   - Improved parsing performance and reduced redundant logic.

---

## Why This PR Targets `main` Branch

The changes here are specifically meant for **v1.3.0**, which is an incremental update from **v1.2.0**. Since no branch exists for `1.3.0` in the base repository, the PR is targeting the `main` branch for now. 

### Suggested Workflow
1. Create a new branch in the base repository:
   ```bash
   git checkout v1.2.0
   git checkout -b release/v1.3.0

### CODE REVIEW

**Comment by jp-netsis** [comment]
Thanks for the pull request.
I understand what you're trying to do.

I am concerned that rich tags have too many features.
The following implementation would still work:
```
<ruby=ice>fire</ruby> // Default
<ruby=ice>fire<color="blue"></ruby></color> // Might ok
<ruby=ice>fire<alpha=#80></ruby><alpha=#FF> // Might ok
<ruby=ice>fire<gradient=Xyz></ruby></gradient> // Might ok
<ruby=ice>fire<b></ruby></b> // Might ok
<ruby=ice>fire<i></ruby></i> // Might ok
<ruby=ice>fire<u></ruby></u> // Might ok
<ruby=ice>fire<s></ruby></s> // Might ok
<ruby=ice>fire<lowercase></ruby></lowercase> // Might ok
<ruby=ice>fire<uppercase></ruby></uppercase> // Might ok
<ruby=ice>fire<smallcaps></ruby></smallcaps> // Might ok
<ruby=ice>fire<mark=#ffff00aa></ruby></mark> // Might ok
<ruby=ice>fire<rotate=45></ruby></rotate> // Might ok, Depending on the amount of rotation, the letters may overlap.
<ruby=ice>fire<font-weight="900"></ruby></font-weight> // Might ok, Depending on the amount of weight, the letters may overlap.
<ruby=ice>fire<cspace=1em></ruby></cspace> // It will probably be reflected, but there is no guarantee that the characters will overlap.
<ruby=ice>fire<mspace=1em></ruby><mspace> // It will probably be reflected, but there is no guarantee that the characters will overlap.
<ruby=i<space=1em>ce>fire</ruby> // This would not be possible with the current implementation
<ruby=ice>fire<size=5></ruby></size> // Depending on the size, the display may be distorted
<ruby=ice>fire<material=2></ruby></material> // Depending on the material, the display may be distorted.
<ruby=ice<sprite="SampleEmoji" name="Xyz">>fire</ruby> // This would not be possible with the current implementation
<ruby=H<sub>2</sub>O>fire</ruby> // This would not be possible with the current implementation
<ruby=i<br>c<br>e>fire</ruby> // This would not be possible with the current implementation
<ruby=ice</align>fire<align="right"></ruby> // There will probably be many problems
<ruby=<align="right">i<align="center">c<align="left">e</align>fire</ruby> // This would not be possible with the current implementation
<ruby=ice>fire<pos=1em></ruby> // There will probably be many problems
```

inndent, line-inndent, voffset, mergin, ...

There are plenty of other tags as well.
Basically, I plan to make it the user's responsibility to use the tags.
I want to think of a better way.

**Comment by kwan3854** [comment]
I completely understand your concerns and appreciate the careful consideration you’ve given to the potential complexities of implementing rich text features.

To clarify, the main motivation for supporting the standard ruby tag is not to expand rich text functionality, but to align with established web standards. This approach ensures better data compatibility and reduces the integration burden for projects that span multiple platforms, such as hybrid web/Unity applications.

For instance, when shared data is received from a server, adhering to a standard format minimizes inconsistencies and simplifies workflows, particularly for developers new to Unity or hybrid development. While the rich text features are indeed a welcome side effect, they are secondary to the goal of improving interoperability and lowering entry barriers for users.

That said, I fully respect your role as the project owner and will completely understand if you decide this change doesn’t align with the repository’s direction. I just wanted to provide additional context to support the discussion.

Thank you again for your time and thoughtful review!

File: .DS_Store
Comment (jp-netsis) [inline .DS_Store:1]: I assume you are a Mac developer.
How about adding `.DS_Store` to `.gitignore`?

File: Assets/RubyTextMeshPro/Source/RubyTextMeshProDefinitions.cs [hunk 1]
```diff
@@ -35,8 +35,24 @@ public static string ReplaceRubyTags(this IRubyText targetRubyText, string str,
             {
                 foreach (Match match in matches)
                 {
-                    if (match.Groups.Count != 5)
+                    string baseText;
+                    string rubyText;
+
+                    if (match.Groups["base"].Success && match.Groups["rubyText"].Success)
+                    {
+                        // 새로운 <ruby>base<rt>value</rt></ruby> 형식
```
Comment (jp-netsis) [inline Assets/RubyTextMeshPro/Source/RubyTextMeshProDefinitions.cs]:
```suggestion
                        // "<ruby>base<rt>value</rt></ruby>" format
```

File: Assets/RubyTextMeshPro/Source/RubyTextMeshProDefinitions.cs [hunk 2]
```diff
@@ -35,8 +35,24 @@ public static string ReplaceRubyTags(this IRubyText targetRubyText, string str,
...
+
+                    if (match.Groups["base"].Success && match.Groups["rubyText"].Success)
+                    {
+                        // 새로운 <ruby>base<rt>value</rt></ruby> 형식
+                        baseText = match.Groups["base"].Value;
+                        rubyText = match.Groups["rubyText"].Value;
+                    }
+                    else if (match.Groups["val"].Success && match.Groups["ruby"].Success)
                     {
+                        // 기존 <ruby="value">base</ruby> 형식
```
Comment (jp-netsis) [inline Assets/RubyTextMeshPro/Source/RubyTextMeshProDefinitions.cs]:
```suggestion
                        // "<ruby=“value”>base</ruby>" format
```

**Review by jp-netsis — CHANGES_REQUESTED** [review]
Thanks for the pull request.
If you do this correspondence, I will merge it.

**Comment by kwan3854** [comment]
> Thanks for the pull request. If you do this correspondence, I will merge it.

I’ve addressed all the suggested changes and updated the branch.

**Review by jp-netsis — APPROVED** [review]
LGTM

**Pull request merged**